packages/corelib/src/dataModel/Rundown.ts

```diff
@@ -83,13 +82,36 @@ export interface Rundown {
 	playlistExternalId?: string
 	/** Whether the end of the rundown marks a commercial break */
 	endOfRundownIsShowBreak?: boolean
-	/** Name (user-facing) of the external NCS this rundown came from */
-	externalNRCSName: string
 	/** The id of the Rundown Playlist this rundown is in */
 	playlistId: RundownPlaylistId
 	/** If the playlistId has ben set manually by a user in Sofie */
 	playlistIdIsSetInSofie?: boolean
 }
 
+export type RundownSource = RundownSourceNrcs | RundownSourceSnapshot | RundownSourceHttp
+
+export interface RundownSourceNrcs {
+	type: 'nrcs'
+	/** The peripheral device the rundown originates from */
+	peripheralDeviceId: PeripheralDeviceId
+	/** Name (user-facing) of the external NCS this rundown came from */
+	nrcsName: string
+}
+export interface RundownSourceSnapshot {
+	type: 'snapshot'
+	rundownId: RundownId
+}
+export interface RundownSourceHttp {
+	type: 'http'
+}
+
+export function getRundownNrcsName(rundown: ReadonlyDeep<Pick<DBRundown, 'source'>> | undefined): string {
+	if (rundown?.source?.type === 'nrcs') {
+		return rundown.source.nrcsName
+	} else {
+		return 'NRCS'
```

Comment:
Should we replace `'NRCS'` with a constant, to use the same as the fallback value in `getExternalNRCSName()`?

Reply:
Instead I've made it so that `nrcsName` is `string | undefined`, which consolidates where we use the `'NRCS'` constant to one usage